### ISSUE TITLE
[Bug] Charging Booster levels do not properly apply cooldown decrease

### ISSUE THREAD

**Issue by ajsglist** [Malcolm-Q/LC-LateGameUpgrades#513]
Hey there,

As per the title, Charging Booster levels beyond the first rank do not properly apply the incremental cooldown value

Expected behavior from these settings would be: 90, 60, 30, 0s

![image](https://github.com/user-attachments/assets/f653602c-d860-49e3-a195-b03d4a177be0)

![image](https://github.com/user-attachments/assets/796bbe93-c247-4812-a4cc-3e27ac4874f4)

It's also unclear what a negative cooldown would do in practice, so it may be worth clamping the minimum to 0s to avoid undesired behavior.

Thanks!

**Comment by ajsglist** [comment]
Protein Powder appears to have similar issues -- the subsequent upgrade is folded into the base value, despite the upgrade seemingly only applying the subsequent upgrade value to L2+ upgrades

![image](https://github.com/user-attachments/assets/d294a19d-f9eb-49a3-9810-18abd6b23aa5)

**Comment by WhiteSpike** [comment]
> As per the title, Charging Booster levels beyond the first rank do not properly apply the incremental cooldown value

They are just incorrectly labelled as in each subsequent entry from the first are assuming a level value incremented by one.
This is just a visual issue, cooldown logic of the booster itself either is also incorrect due to a different factor or it remains correct.
I will be forced to worry about this after the two week break I have planned from this Sunday.

> It's also unclear what a negative cooldown would do in practice, so it may be worth clamping the minimum to 0s to avoid undesired behavior.

It's already clamped.

> Protein Powder appears to have similar issues -- the subsequent upgrade is folded into the base value, despite the upgrade seemingly only applying the subsequent upgrade value to L2+ upgrades

Each level describes what you have **total** at that level, not what you will have additionally when reaching that level.
As in:
At level one, you will have an effect of "startingAmount"
At level two, you will have an effect of "startingAmount" + "level2Amount"
At level three, you will have an effect of "startingAmount" + "level2Amount" + "level3Amount".

That is the premise from which the information related to each upgrades was built on.

**Comment by ajsglist** [comment]
>Each level describes what you have total at that level, not what you will have additionally when reaching that level.
>As in:
>At level one, you will have an effect of "startingAmount"
>At level two, you will have an effect of "startingAmount" + "level2Amount"
>At level three, you will have an effect of "startingAmount" + "level2Amount" + "level3Amount".

Right.  Currently the L1 value erroneously shows startingAmount + level2Amount, compounding the issue for all further levels.

**Comment by WhiteSpike** [comment]
> Right. Currently the L1 value erroneously shows startingAmount + level2Amount, compounding the issue for all further levels.

If I remember correctly from Protein, no. It should be "baseShovelDamage + startingAmount" as the format.
I guess it can be argued that base damage from the shovel-like items shouldn't be considered into the upgrade's description and I can just this one increment or just rephrase it to make it so it's actually setting how much damage you deal.

**Comment by ajsglist** [comment]
>I guess it can be argued that base damage from the shovel-like items shouldn't be considered into the upgrade's description and I can just remove this one increment from it or just rephrase it to make it so it's actually setting how much damage you deal.

Would be greatly appreciated when you find the time.

Thanks again.  Enjoy your break! :)
